src/test/tests.ts

```diff
@@ -234,6 +241,31 @@ describe('SubscriptionManager', function() {
     });
   });
 
+  it('can use an async filter function', function(done) {
```

Comment:
oh no, i don't mind at all that this is what you are testing.
i would just rename the test abit to something like "can use promise inside filter function"
or something like that.. since usually when testing 'async', the testing is about being run in parallel